File: apps/api/proxima_api/event_types.py
"""Canonical names for durable session-stream events.

The database remains the event source of truth. These registries keep producers
and clients aligned without turning event names into control authority.
"""
from __future__ import annotations

MASTER_TASK_EVENT_TYPES = frozenset(
    {
        "master.task.started",
        "master.task.completed",
        "master.task.failed",
        "master.task.cancelled",
        "master.task.review_ready",
        "master.task.blocked",
    }
)

MASTER_TASK_RECOVERY_EVENT_TYPES = frozenset(
    {
        "master.task.recovered",
        "master.task.recovery_history_corrected",
    }
)

MASTER_SUPERVISION_EVENT_TYPES = frozenset(
    {
        "master.attention.required",
        "master.decision.deferred",
        "master.decision.resolved",
        "master.supervisor.outcome",
        "master.satpam.steered",
        "master.satpam.restart_queued",
        "master.satpam.restarted",
        "master.satpam.recovery_failed",
        "master.satpam.escalated",
    }
)

MASTER_PROJECTION_EVENT_TYPES = (
    MASTER_TASK_EVENT_TYPES | MASTER_SUPERVISION_EVENT_TYPES
)

MASTER_FOCUS_EVENT_TYPES = frozenset({"master.focus.changed"})

GRAPH_STATE_EVENT_TYPES = frozenset(
    {
        "graph.state.missing",
        "graph.state.queued",
        "graph.state.building",
        "graph.state.fresh",
        "graph.state.stale",
        "graph.state.failed",
    }
)

File: apps/web/src/lib/eventTypes.ts
// Canonical named events delivered by the existing session SSE contract.
// Keep this list aligned with backend event_types.py when adding projections.
export const SESSION_EVENT_TYPES = [
  'run.queued', 'run.started', 'message.delta', 'reasoning.delta', 'tool.start', 'tool.complete',
  'approval.request', 'approval.auto', 'approval.denied', 'message.complete', 'run.completed', 'run.failed', 'run.cancelled',
  'warning', 'wiki.draft', 'workflow.draft', 'goal.update',
  'collaboration.child.queued', 'collaboration.child.started', 'collaboration.child.delta',
  'collaboration.child.completed', 'collaboration.child.failed', 'collaboration.child.cancelled',
  'message_review.queued', 'message_review.started', 'message_review.completed',
  'message_review.failed', 'message_review.applied', 'message_review.restored',
  'graph.node.update', 'job.update',
  'satpam.steered', 'satpam.restart.queued', 'satpam.restarted', 'satpam.escalated',
  'master.task.started', 'master.task.completed', 'master.task.failed',
  'master.task.cancelled', 'master.task.review_ready', 'master.task.blocked',
  'master.task.recovered',
  'master.attention.required', 'master.supervisor.outcome',
  'master.satpam.steered', 'master.satpam.restart_queued',
  'master.satpam.restarted', 'master.satpam.recovery_failed',
  'master.satpam.escalated',
  'master.focus.changed',
  'graph.state.missing', 'graph.state.queued', 'graph.state.building',
  'graph.state.fresh', 'graph.state.stale', 'graph.state.failed',
] as const

export type SessionEventType = typeof SESSION_EVENT_TYPES[number]
